### PR TITLE
chore: migrate to native test suites

### PR DESCRIPTION
**Pull request by clemstoquart**
Hi,

I've draft the migration to the jvm test suite plugin. But I'm not satisfied with the way we handle dependencies versions.

Any idea is very welcome.

First step for https://github.com/ekino/spring-boot-starter-errors/issues/256

### CODE REVIEW

**Comment by clemstoquart** [comment]
IMO it's kinda acceptable in this state. I've not manage to have something working with the version catalog.

@celcius112 What do you think ?

File: build.gradle.kts [hunk 1]
```diff
@@ -30,23 +28,80 @@ dependencies {
...
 
-  securityTestImplementation("org.springframework.security:spring-security-web")
+        val securityTest by registering(JvmTestSuite::class) {
+            dependencies {
+                implementation(project())
+                implementation("org.springframework.security:spring-security-web")
+            }
+            sources {
+                java {
+                    setSrcDirs(listOf("src/securityTest/kotlin"))
```
Comment (celcius112) [inline build.gradle.kts:59]: is this necessary ? i think by default it will use the task name

File: build.gradle.kts [hunk 2]
```diff
@@ -30,23 +28,80 @@ dependencies {
...
-configurations {
-  all {
-    exclude(module = "mockito-core")
-    exclude(module = "assertj-core")
-  }
+tasks.named("build") {
+    dependsOn(testing.suites.named("securityTest"))
+    dependsOn(testing.suites.named("dataRestTest"))
+    dependsOn(testing.suites.named("txTest"))
+    dependsOn(testing.suites.named("awsTest"))
```
Comment (celcius112) [inline build.gradle.kts:106]: what would you think of `testing.suites.forEach(::dependsOn)` instead ?

Comment (clemstoquart) [reply build.gradle.kts]: Nice catch, done.

**Review by celcius112 — APPROVED** [review]

**Pull request merged**